app/components/provider_interface/filter_component.rb

```diff
@@ -2,24 +2,64 @@ module ProviderInterface
   class FilterComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :params_for_current_state, :available_filters, :applied_filters
+    attr_reader :page_state
+    delegate :filters, to: :page_state
 
-    def initialize(available_filters:, applied_filters:, params_for_current_state:)
-      @available_filters = available_filters
-      @applied_filters = applied_filters
-      @params_for_current_state = params_for_current_state
+    def initialize(page_state:)
+      @page_state = page_state
     end
 
-    def checkbox_checked?(heading:, name:)
-      applied_filters.dig(heading, name) ? true : false
+    def tags_for_active_filter(filter)
+      case filter[:type]
+      when :search
+        [{ title: filter[:value], remove_link: remove_search_tag_link(filter[:name]) }]
+      when :checkboxes
+        filter[:options].each_with_object([]) do |option, arr|
+          if option[:checked]
+            arr << { title: option[:label], remove_link: remove_checkbox_tag_link(filter[:name], option[:value]) }
+          end
+        end
+      end
     end
 
-    def is_candidates_name_search_field?(filter_group)
-      filter_group[:heading].parameterize.eql?('candidate-s-name')
+    def filter_should_render?(filter_options:)
```

Comment:
Lovely method name! But... it looks generic when it's not. If there are no options for the filter in question, should it even be added to the filters array in the first place?

Reply:
Good point, but thinking about it now, this method is incorrect. It is not that it is less than 1 that we should care about, but that it is be that it is less than 2. If it is less than two it means that there is only one possible filter so it is redundant as it would include all of the applications on that page. This was the original intention for this check.

For instance if there is only 1 provider - then all applications would belong to it, there fore filtering for it would just bring back everything in any case.

Does that make sense?

Reply:
sorry — I see this is for the case where there's only one option

Reply:
Have moved logic to page_state and updated specs as discussed 👍 see: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2114/commits/af1312ba73bcb5b081e07a402cc7f4bce4caff30